wurst/util/GroupUtils.wurst

```diff
@@ -0,0 +1,134 @@
+package GroupUtils
+import NoWurst
+import Group
+import ErrorHandling
+import HashMap
+import Execute
+import MagicFunctions
+import TypeCasting
+
+/*
+    Groups are a necessary evil that exists in warcraft 3, and sometimes their use
+    cannot be easily replicated or avoided, especially with nested enums.
+
+    However, for most purposes, if you want to keep a list of units and iterate over them
+    often, you should use LinkedList<unit> instead of the group, since LinkedList<unit>
+    provides more predictable and consistent functionalities that can be extended more easily.
+*/
+
+/** The max number of recyclable groups. Even if this number is reached, 
+    the system will create new groups on following requests, but they won't be recyclable.
+    The maximum number for this setting is 8191. */
+@configurable constant GROUP_NUMBER_LIMIT = 1024
+
+/** The number of groups created when more are needed */
+@configurable constant NEED_CREATE_GROUPS = 64
+
+/** The number of groups created at start */
+@configurable constant START_CREATE_GROUPS = 64
+
+/* Group instantiation */
+
+/** The total number of recyclable groups created by GroupUtils */
+
+let used = new HashMap<group, bool>
+group array stack
+var numStack = 0
+var numTotal = 0
+
+var shownMaxError = false
+
+function createGroups(int number)
+    let maxCreatePerCycle = 256
+    let actualLimit = IMinBJ(8191, GROUP_NUMBER_LIMIT)
+    let numTarget = numTotal + number 
+    let overflow  = IMaxBJ(0, numTarget - actualLimit)
+    var toCreate  = IMaxBJ(0, numTarget - overflow)
+
+    while toCreate > 0
+        let createNow = IMinBJ(toCreate, maxCreatePerCycle)
+        toCreate -= createNow
+
+        if compiletime 
```

Comment:
one should just implement ForForce for compiletime instead of this conditional imo.